token/link/github.go

```diff
@@ -1,45 +1,65 @@
 package link
 
 import (
+	"context"
+	"encoding/json"
 	"strings"
 
 	"github.com/fabric8-services/fabric8-auth/client"
+	"github.com/fabric8-services/fabric8-auth/token/oauth"
 
 	"github.com/satori/go.uuid"
 	"golang.org/x/oauth2"
 	"golang.org/x/oauth2/github"
 )
 
 const (
-	gitHubProviderID = "2f6b7176-8f4b-4204-962d-606033275397" // Do not change! This ID is used as provider ID in the external token table
+	GitHubProviderID = "2f6b7176-8f4b-4204-962d-606033275397" // Do not change! This ID is used as provider ID in the external token table
 )
 
-type GitHubConfig struct {
-	oauth2.Config
-	providerID uuid.UUID
-	scopeStr   string
+type GitHubIdentityProvider struct {
+	oauth.OauthIdentityProvider
 }
 
-func NewGitHubConfig(clientID string, clientSecret string, scopes string, authURL string) *GitHubConfig {
-	provider := &GitHubConfig{}
+type gitHubUser struct {
+	Login string `json:"login"`
+}
+
+func NewGitHubIdentityProvider(clientID string, clientSecret string, scopes string, authURL string) *GitHubIdentityProvider {
+	provider := &GitHubIdentityProvider{}
 	provider.ClientID = clientID
 	provider.ClientSecret = clientSecret
 	provider.Endpoint = github.Endpoint
 	provider.RedirectURL = authURL + client.CallbackTokenPath()
-	provider.scopeStr = scopes
+	provider.ScopeStr = scopes
 	provider.Config.Scopes = strings.Split(scopes, " ")
-	provider.providerID, _ = uuid.FromString(gitHubProviderID)
+	provider.ProviderID, _ = uuid.FromString(GitHubProviderID)
+	provider.ProfileURL = "https://api.github.com/user"
```

Comment:
Should be in configuration.

Reply:
I don't think so. This is GitHub. It's not supposed to be configurable in different environments.